src/h3lib/lib/h3Index.c

```diff
@@ -31,6 +31,40 @@
 #include "iterators.h"
 #include "mathExtensions.h"
 
+/** @var H3ErrorDescriptions
+ *  @brief An array of strings describing each of the H3ErrorCodes enum values
+ */
+static char *H3ErrorDescriptions[] = {
+    "Success",
+    "The operation failed but a more specific error is not available",
+    "Argument was outside of acceptable range",
+    "Latitude or longitude arguments were outside of acceptable range",
+    "Resolution argument was outside of acceptable range",
+    "Cell argument was not valid",
+    "Directed edge argument was not valid",
+    "Undirected edge argument was not valid",
+    "Vertex argument was not valid",
+    "Pentagon distortion was encountered",
+    "Duplicate input",
+    "Cell arguments were not neighbors",
+    "Cell arguments had incompatible resolutions",
+    "Memory allocation failed",
+    "Bounds of provided memory were insufficient",
+    "Mode or flags argument was not valid"};
+
+/**
+ * Returns the string describing the H3Error
+ * @param err The H3 error.
+ * @return The string describing the H3Error
+ */
+char *H3_EXPORT(describeH3Error)(H3Error err) {
```

Comment:
```suggestion
const char *H3_EXPORT(describeH3Error)(H3Error err) {
```
I'm trying to follow the API used by e.g. SDL2 https://wiki.libsdl.org/SDL2/SDL_GetError -- modifying the string returned by this function would probably be bad. We will need to add similar instructions from that page that the returned memory should not be freed, etc.

(edit: note the sqlite interface is very similar https://www.sqlite.org/c3ref/errcode.html)

Reply:
Let me know if the extra comment is good enough!